scripts/s3-support-bundles/s3_bundle_generate.sh

```diff
@@ -31,18 +31,32 @@ where:
 -c        Confstore URL.
 -s        services list."
 
+SHORT=b:,t:,c:,s:
+LONG=duration:,size_limit:,binlogs:,coredumps:,stacktrace:
+
 usage() {
   echo "$USAGE"
   exit 1
 }
 
-while getopts "b:t:c:s:" opt; do
-  case "${opt}" in
-    b) bundle_id=${OPTARG} ;;
-    t) bundle_path=${OPTARG} ;;
-    c) confstore_url=${OPTARG} ;;
-    s) services=${OPTARG} ;;
-    *) usage ;;
+OPTS=$(getopt --options $SHORT --long $LONG  --name "$0" -- "$@")
+
+eval set -- "$OPTS"
+
+# extract options and their arguments into variables.
+while true ; do
+  case "$1" in
+    -b) bundle_id="$2";shift 2;;
+    -t) bundle_path="$2";shift 2;;
+    -c) confstore_url="$2";shift 2;;
+    -s) services="$2";shift 2;;
+    --duration) duration="$2";shift 2;;
+    --size_limit) size_limit="$2";shift 2;;
+    --binlogs) binlogs="$2";shift 2;;
+    --coredumps) coredumps="$2";shift 2;;
+    --stacktrace) stacktrace="$2";shift 2;;
+    --) shift;break;;
+    *) usage;exit 1;;
```

Comment:
Update usage @shahidSTX

Reply:
Also - 'where are we differentiating 'mandatory' and 'optional' parameters

Reply:
The check for mandatory parameters remains the same, using -z flag.

Reply:
Usage updated.